givelively/exception_notifier
# Du/dev 1160/fix capture message

[DEV-1160](https://give-lively.atlassian.net/browse/DEV-1160)

If you call `ExceptionNotifier.call` with a string as the first argument, we call `ExceptionNotifier.capture_message` internally. It seems like the intention is to create a Sentry issue with the given name, e.g. if you do `ExpectionNotifier.call("My Sentry issue", occurrence: 436)`, it should show up in Sentry as "My Sentry issue."

 However, because we use the splat operator (`*args`) twice on the same arg list, once in `call` and once in `capture_message`, the args end up as an array inside an array, so `args.first.is_a?(String)` is always false. This results in all messages getting reported to Sentry as "Unknown parameter set."

This PR removes the second splat, so that the string argument is used as the name of the Sentry issue. It also fixes the way we handle the other arguments in `capture_message`—if there's only one other argument and it's a hash, we pass the hash along to Sentry; otherwise, we send the rest of the argument array as `parameters`.

## QA Notes

#### Steps to QA



1. Set `SENTRY_DSN` in your .env file to [the value seen here](https://sentry.io/settings/give-lively/projects/charity-api/keys/)
1. Remove `gem 'exception_notifier', '0.1.2', git: 'https://github.com/givelively/exception_notifier/'` from your Gemfile in charity-api
2. Replace it with `gem 'exception_notifier', path: '../exception_notifier'`
3. Run `bundle install`
4. Open up a Rails console
5. Run `ExceptionNotifier.call('Some test issue', foo: 'bar')`
6. View the exception in [Sentry](https://sentry.io/organizations/give-lively/issues/?project=2530079); you should be able to search for it as "Some test issue"
7. Scroll down to "Additional Data"
8. Make sure `foo` is set to `bar`

When you're done testing this, remember to delete `SENTRY_DSN` from your .env file so we don't clutter Sentry with our local exceptions.
